fetch-in-react-practice/client/components/app.jsx

```diff
@@ -76,29 +76,26 @@ export default class App extends React.Component {
      * TIP: Be sure to SERIALIZE the updates in the body with JSON.stringify()
      * And specify the "Content-Type" header as "application/json"
      */
-    const todosArray = this.state.todos;
-    for (var i = 0; i < todosArray.length; i++) {
-      if (todosArray[i].todoId === todoId) {
-        const newObject = todosArray[i];
-        newObject.isCompleted = !newObject.isCompleted;
-      }
-    }
-
-    /*  fetch('/api/todos/{todoId}', {
+    const todoIndex = this.state.todos.findIndex(item => item.todoId === todoId);
+    const todoObject = this.state.todos[todoIndex];
+    todoObject.isCompleted = !todoObject.isCompleted;
```

Comment:
For this part, we will want to make a new object instead of modifying the original.